LBH.AdultSocialCare.Api/V1/UseCase/Payments/Concrete/CreateDraftPayRunUseCase.cs

```diff
@@ -28,29 +27,48 @@ public CreateDraftPayRunUseCase(IPayRunGateway payRunGateway, IQueueService payr
 
         public async Task CreateDraftPayRun(DraftPayRunCreationDomain draftPayRunCreationDomain)
         {
-            var draftPayRunCount = await _payRunGateway.GetDraftPayRunCount(draftPayRunCreationDomain.Type);
-            if (draftPayRunCount > 0)
-                throw new ApiException($"A Pay Run with draft status for {draftPayRunCreationDomain.Type.ToDescription()} already exists!");
-
+            var recurringPayRunTypes = new[] { PayrunType.DirectPayments, PayrunType.ResidentialRecurring };
             ValidateDraftPayRun(draftPayRunCreationDomain);
 
+            var unApprovedPayRunExists = await _payRunGateway.CheckExistsUnApprovedPayRunAsync(draftPayRunCreationDomain.Type);
+            if (unApprovedPayRunExists)
```

Comment:
This condition is valid only for recurring, not AdHoc. It is possible two payruns (one for recurring, one for Adhoc)

For Adhoc, you should check startDate and endDate. There are more than one ad-hoc payruns if it is not overlapped. 

For example; 
Valid case
AdHocPayrun1.Date: 01.10.2021 - 15.10.2021 (Draft)
AdHocPayrun2.Date: 16.10.2021 - 30.10.2021 (Draft)
AdHocPayrun3.Date: 01.11.2021 - 30.11.2021 (Draft)

Invalid case
AdHocPayrun1.StartDate: 01.10.2021 - 15.10.2021 (Draft)
AdHocPayrun2.StartDate: 13.10.2021 - 30.10.2021 (Invalid because overlapped date)

Reply:
Ad-hoc payruns for released invoices can actually overlap (I think). Or some held->released invoices will be skipped

Reply:
That's right. Adhoc can be overlap with Regular cycle. Let me try to explain shortly (I hope :) ); 

- Payrun cannot be overlapped for same type.
- It is not possible to create regular cycle payrun if there is any draft "payrun".
- It is possible more than one draft payrun if it is not overlapped.